src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PEParameterSymbol.vb

```diff
@@ -655,5 +655,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
                 Return DirectCast(_containingSymbol.ContainingModule, PEModuleSymbol).Module
             End Get
         End Property
+
+        Friend Function DeriveCompilerFeatureRequiredDiagnostic(decoder As MetadataDecoder) As DiagnosticInfo
+            Return DeriveCompilerFeatureRequiredAttributeDiagnostic(Me, DirectCast(Me.ContainingModule, PEModuleSymbol), Handle, CompilerFeatureRequiredFeatures.None, decoder)
+        End Function
+
+        Public Overrides ReadOnly Property HasUnsupportedMetadata As Boolean
+            Get
+                Dim containingModule = DirectCast(Me.ContainingModule, PEModuleSymbol)
+                Dim decoder = If(TypeOf ContainingSymbol Is PEMethodSymbol,
```

Comment:
>TypeOf ContainingSymbol Is PEMethodSymbol [](http://example.com/codeflow?start=33&length=41)

It is probably more efficient to do TryCast to a local and then checking for null before usage.